circle/src/deep_quotient.rs

```diff
@@ -0,0 +1,101 @@
+use alloc::vec::Vec;
+
+use itertools::{izip, Itertools};
+use p3_commit::PolynomialSpace;
+use p3_field::extension::{Complex, ComplexExtendable, ExtensionPowersReducer};
+use p3_field::{batch_multiplicative_inverse, ExtensionField};
+use p3_matrix::dense::RowMajorMatrix;
+use p3_matrix::Matrix;
+use p3_maybe_rayon::prelude::*;
+use p3_util::log2_strict_usize;
+use tracing::instrument;
+
+use crate::domain::CircleDomain;
+use crate::util::v_n;
+
+/// Compute numerator and denominator of the "vanishing part" of the DEEP quotient
+/// Section 6, Remark 21 of Circle Starks (page 30 of first edition PDF)
+/// Re(1/v_gamma) + alpha^L Im(1/v_gamma)
+/// (Other "part" is \bar g - \bar v_gamma)
+pub(crate) fn deep_quotient_vanishing_part<F: ComplexExtendable, EF: ExtensionField<F>>(
+    x: Complex<F>,
+    zeta: Complex<EF>,
+    alpha_pow_width: EF,
+) -> (EF, EF) {
+    let x_rotate_zeta: Complex<EF> = x.rotate(zeta.conjugate());
+
+    let v_gamma_re: EF = EF::one() - x_rotate_zeta.real();
+    let v_gamma_im: EF = x_rotate_zeta.imag();
```

Comment:
This should probably be either another function which computes v_gamma from x and zeta or at least add a comment about where this formula is coming from. (I think it's equation 11 on the bottom of page 11 from the paper: https://eprint.iacr.org/2024/278.pdf)

Reply:
Also I think a negative has gone missing. According to the definition of v on page 11, im(v(x, y)) = -y not y.

This could also be fixed by replacing v_gamma_re - alpha_pow_width * v_gamma_im below by v_gamma_re + alpha_pow_width * v_gamma_im

Reply:
yep, had that wrong, thx